test/plugins/smartQuotes.test.ts

```diff
@@ -0,0 +1,993 @@
+import { Configuration } from '../testConfiguration';
+import { newTest } from '../testSimplifier';
+import { cleanUpWorkspace, setupWorkspace } from '../testUtils';
+
+suite('smartQuotes plugin', () => {
```

Comment:
Love to see thorough tests like this; thanks! 🙂

Reply:
Tried to cover everything.
Though did I miss a few cases and added them later, using the same code I where I found the bug 🙃